services/provider/server/server.go

```diff
@@ -1052,10 +1066,11 @@ func (s *OCSProviderServer) getKubeResources(ctx context.Context, consumer *ocsv
 	}
 
 	kubeResources := []client.Object{}
-	kubeResources, err = s.appendCephConnectionKubeResources(ctx, kubeResources, consumer)
+	cephConnection, err := s.getDesiredCephConnection(ctx, consumer)
 	if err != nil {
 		return nil, err
 	}
+	kubeResources = append(kubeResources, cephConnection)
```

Comment:
More structured and limits the scope of the ceph connection object
```suggestion
	if cephConnection, err := s.getDesiredCephConnection(ctx, consumer); err == nil {
		kubeResources = append(kubeResources, cephConnection)
	} else {
		return nil, err
	}

```